src/Symfony/Bundle/SecurityBundle/Command/UserPasswordEncoderCommand.php

```diff
@@ -86,13 +89,18 @@ protected function execute(InputInterface $input, OutputInterface $output)
         }
 
         if (!$salt) {
-            $saltQuestion = $this->createSaltQuestion($input, $output);
-            $salt = $helper->ask($input, $output, $saltQuestion);
+            if ($input->isInteractive()) {
+                $saltQuestion = $this->createSaltQuestion($input, $output);
+                $salt = $helper->ask($input, $output, $saltQuestion);
+            } else {
+                $salt = $this->generateSalt($output);
+            }
+
         }
 
         $output->writeln("\n <comment>Encoders are configured by user type in the security.yml file.</comment>");
 
-        if (!$userClass) {
+        if (!$userClass && $input->isInteractive()) {
             $userClassQuestion = $this->createUserClassQuestion($input, $output);
```

Comment:
What I mean by "useless" is what I explained the #13988 :
> Since we configure Symfony\Component\Security\Core\User\User in the setAutocompleterValues and set the $value to Symfony\Component\Security\Core\User\User if null, I guess we can configure the default value of the user-class.

Indeed, why not providing the Symfony\Component\Security\Core\User\User as the default value of the user-class option/argument ?
Asking it explicitly to the user is only useful for new symfony's users. Once understood, I think most users will provide it directly through the `user-class` option, or expect the default Symfony\Component\Security\Core\User\User value.

The only point of this question would be if the user was able to pick one encoder from a choice list. But as we cannot access the configured encoders, this question doesn't have any value to me. :/